pkg/sql/conn_executor_jobs.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/errors"
+)
+
+func (ex *connExecutor) waitOneVersionForNewVersionDescriptorsWithoutJobs(
```

Comment:
Give this a comment on why we want to do it / why it matters

Reply:
done.